File: backend/internal/validation/phone.go
package validation

import (
	"errors"
	"regexp"
)

// ErrInvalidPhone is returned when a phone number doesn't match the Sri Lankan format.
var ErrInvalidPhone = errors.New("phone number must be a valid Sri Lankan number (e.g. 0771234567, 0112345678, or +94771234567)")

// sriLankanPhone matches a leading +94 or 0 followed by 9 digits, covering both mobile numbers and landlines with a 2-digit area code.
var sriLankanPhone = regexp.MustCompile(`^(?:\+94|0)\d{9}$`)

// IsValidSriLankanPhone reports whether s is empty (optional fields) or a validly formatted Sri Lankan phone number.
func IsValidSriLankanPhone(s string) bool {
	if s == "" {
		return true
	}
	return sriLankanPhone.MatchString(s)
}

File: frontend/src/shared/lib/phone.ts
// +94 or 0 followed by 9 digits covers Sri Lankan mobiles and landlines; mirrors backend internal/validation/phone.go.
const SRI_LANKAN_PHONE = /^(?:\+94|0)\d{9}$/;

// Empty is valid because most phone fields are optional; callers add their own presence check.
export function isValidSriLankanPhone(value: string): boolean {
  const trimmed = value.trim();
  if (trimmed === "") return true;
  return SRI_LANKAN_PHONE.test(trimmed);
}

export const PHONE_INVALID_TEXT =
  "Enter a valid Sri Lankan number, e.g. 0771234567, 0112345678, or +94771234567.";
